src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/ErrorNotificationService.java

```diff
@@ -30,10 +43,24 @@ public void processServiceBusMessage(ErrorMsg message) {
             message.errorDescription,
             message.id
         );
+        ErrorNotification entity = new ErrorNotification(
+            entityManager.getReference(ProcessEvent.class, message.eventId),
+            message.errorCode.name(),
+            message.documentControlNumber,
+            message.id
+        );
+
+        try {
+            ErrorNotificationResponse response = client.notify(request);
+
+            entity.setNotificationId(response.getNotificationId());
 
-        ErrorNotificationResponse response = client.notify(request);
+            log.info("Error notification published. ID: {}", response.getNotificationId());
+        } catch (Exception exception) {
+            log.error("Failed to publish error notification. {}", exception.getMessage(), exception);
+        }
 
-        // store result in db. wip
-        log.info("Error notification published. ID: {}", response.getNotificationId());
+        // save the entity after everything is processed
+        repository.save(entity);
```

Comment:
Should this be inside the `try` clause?

Reply:
possibly. we don't want to dead letter the notification just because we failed to save the fact

Reply:
There is more to the story. The way message needs to be handled differs from the response received from client/service. Will introduce some extra error handling in another PR. It should cover all edge cases and conclude the story. Not sure whether I will finish it but I'll liaise with you anyway